docs/source/common/lightning_cli.rst

```diff
@@ -929,6 +957,25 @@ You can also pass the class path directly, for example, if the optimizer hasn't
         --gen_discriminator.init_args.lr=0.0001
 
 
+Troubleshooting
+^^^^^^^^^^^^^^^
+
+The standard behavior for CLIs when they fail is to terminate the process with a non-zero exit code and a short message
+to hint the user about the cause. This is problematic while developing the CLI since there is not information to track
+down the root of the problem. A simple change in the instantiation of the ``LightningCLI`` can be used such that when
+there is a failure an exception is raised and the full stack trace printed.
+
+.. testcode::
+
+    cli = LightningCLI(MyModel, parser_kwargs={"error_handler": None})
+
+.. note::
+
+    When asking about problems and reporting issues please set the ``error_handler`` to ``None`` and include the stack
+    trace in you description. With this it is more likely for people to help out identifying the cause without needing
+    to create a reproducible script.
```

Comment:
```suggestion
The standard behavior for CLIs, when they fail, is to terminate the process with a non-zero exit code and a short message
to hint the user about the cause. This is problematic while developing the CLI since there is no information to track
down the root of the problem. A simple change in the instantiation of the ``LightningCLI`` can be used such that when
there is a failure an exception is raised and the full stack trace printed.

.. testcode::

    cli = LightningCLI(MyModel, parser_kwargs={"error_handler": None})

.. note::

    When asking about problems and reporting issues please set the ``error_handler`` to ``None`` and include the stack
    trace in your description. With this, it is more likely for people to help out identifying the cause without needing
    to create a reproducible script.
```